File: docs/src/python/user-guide/concepts/streaming.py
import polars as pl

# --8<-- [start:query]
q = (
    pl.scan_csv("docs/data/iris.csv")
    .filter(pl.col("sepal_length") > 5)
    .group_by("species")
    .agg(pl.col("sepal_width").mean())
)
# --8<-- [end:query]

# --8<-- [start:streaming_query_plan]
import polars as pl

q = (
    pl.scan_csv("docs/data/iris.csv")
    .filter(pl.col("sepal_length") > 5)
    .group_by("species")
    .agg(pl.col("sepal_width").mean())
)
q.explain(streaming=True)
# --8<-- [end:streaming_query_plan]


# --8<-- [start:streaming]
df = q.collect(streaming=True)
# --8<-- [end:streaming]

# # --8<-- [start:non_streaming_query]
# q_shift = (
#     pl.scan_csv("docs/data/iris.csv")
#     .with_columns(pl.col("sepal_length").shift(1))
#     .group_by("species")
#     .agg(pl.col("sepal_width").mean())
# )

# # --8<-- [end:non_streaming_query]
# # --8<-- [start:non_streaming_query_plan]
# q_shift.explain(streaming=True)
# # --8<-- [end:non_streaming_query_plan]

File: docs/src/rust/user-guide/concepts/streaming.rs
use polars::prelude::*;

fn main() -> Result<(), Box<dyn std::error::Error>> {
    // --8<-- [start:query]
    let q = LazyCsvReader::new("docs/data/iris.csv")
        .has_header(true)
        .finish()?
        .filter(col("sepal_length").gt(lit(5)))
        .group_by(vec![col("species")])
        .agg([col("sepal_width").mean()]);
    // --8<-- [end:query]
    # --8<-- [start:streaming_query_plan]
    plan = q.with_streaming(true).describe_plan();
    println!("{}", plan);

    # --8<-- [end:streaming_query_plan]
    
    // --8<-- [start:streaming]

    let df = q.with_streaming(true).collect()?;
    println!("{}", df);
    // --8<-- [end:streaming]

    Ok(())
}
